app/src/main/java/org/metabrainz/android/presentation/features/search/SearchActivity.kt

```diff
@@ -164,14 +170,24 @@ class SearchActivity : AppCompatActivity(), SearchView.OnQueryTextListener {
     }
 
     override fun onQueryTextSubmit(query: String): Boolean {
+        if (!App.context!!.isOnline){
+            Toast.makeText(this,"Connect to Internet and Try Again",Toast.LENGTH_LONG).show()
+            return false
+        }
         saveSearchSuggestion(query)
 
         adapter = ResultPagingAdapter(ResultItemComparator(), searchTypeFromSpinner!!)
         binding.recyclerView.adapter = adapter
         adapter!!.resetAnimation()
-        viewModel!!.search(searchTypeFromSpinner, query).observe(this, { pagingData: PagingData<ResultItem> ->
+        viewModel!!.search(searchTypeFromSpinner, query).observe(this) { pagingData: PagingData<ResultItem> ->
             adapter!!.submitData(lifecycle, pagingData)
-        })
+        }
+        lifecycleScope.launch {
+            adapter!!.loadStateFlow.collect {
+                binding.loadingAnimation.root.isVisible = it.refresh is LoadState.Loading
```

Comment:
What after this? What if the user wants to search for something else? This view hides the entire screen to search. I suggest we add the loading view only along with the size of the recycerView. Another solution could be to undo this view upon pressing the back button.

Reply:
I meant to tag the noResult line which is one line below this!

Reply:
Sorry, I actually started working on it asap so didn't see the second reply.
I actually have now overridden the onBackPressed function, so the user can navigate back to the initially looking fragment,
Also, I made changes in the XML of the loading and noresult animations and set them in a way that they remain below the search bar. 
Please suggest if I can change this approach.